nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-services/src/test/java/org/apache/nifi/marklogic/controller/DefaultMarkLogicDatabaseClientServiceIT.java

```diff
@@ -44,8 +47,16 @@ public void testService() throws InitializationException {
         runner.setProperty(service, DefaultMarkLogicDatabaseClientService.USERNAME, username);
         runner.setProperty(service, DefaultMarkLogicDatabaseClientService.PASSWORD, password);
         runner.setProperty(service, DefaultMarkLogicDatabaseClientService.SECURITY_CONTEXT_TYPE, authentication);
+        runner.setProperty(service, DefaultMarkLogicDatabaseClientService.LOAD_BALANCER, loadBalancer);
         runner.enableControllerService(service);
         runner.assertValid(service);
+        runner.disableControllerService(service);
+        try {
+            service.getDatabaseClient().newBinaryDocumentManager().read("/test.json").getPageSize();
+            Assert.assertTrue("Should not be able to use client after shutdown", false);
+        } catch (Exception e) {
+            Assert.assertTrue("Should not be able to use client after shutdown", e instanceof IllegalStateException);
```

Comment:
So it's okay if an `IllegalStateException` gets thrown?

Reply:
Yes. That is the expected error if the database client has been released and an attempt is made to use the client.